ports/rp2/mbedtls/mbedtls_port.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <py/mpconfig.h>
+
+#ifdef MICROPY_SSL_MBEDTLS
+
+#include "py/runtime.h"
+#include "py/gc.h"
+#include "mbedtls_config.h"
+
+#define DEBUG (0)
+
+extern uint8_t rosc_random_u8(size_t cycles);
+
+#if DEBUG
+static size_t count_links(uint32_t *nb) {
+    void **p = MP_STATE_PORT(mbedtls_memory);
+    size_t n = 0;
+    *nb = 0;
+    while (p != NULL) {
+        ++n;
+        *nb += gc_nbytes(p);
+        p = (void **)p[1];
+    }
+    return n;
+}
+#endif
+
+void *m_calloc_mbedtls(size_t nmemb, size_t size) {
```

Comment:
This malloc code is copied from mimxrt/stm32; we should move it somewhere shared.  Let me do that separately to consolidate mimxrt and stm32.

Reply:
That's a good suggestion, I'll leave it as-is for now.

Reply:
I've made this change on master, so you can now use it